src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -530,3 +543,32 @@ export const performEncryptPostSubmissionActions = (
       })
     })
 }
+
+export const getQuarantinePresignedPostData = (
+  attachmentSizes: AttachmentSizeMapType[],
+): ResultAsync<
+  AttachmentPresignedPostDataMapType[],
+  CreatePresignedPostError
+> => {
+  const totalAttachmentSizeLimit = fileSizeLimitBytes(FormResponseMode.Encrypt)
+  let totalAttachmentSize = 0
+  return ResultAsync.combine(
+    attachmentSizes.map(({ id, size }) => {
+      if (!mongoose.isValidObjectId(id))
+        return errAsync(new InvalidFieldIdError())
+
+      totalAttachmentSize += size
+      if (totalAttachmentSize > totalAttachmentSizeLimit)
+        return errAsync(new AttachmentSizeLimitExceededError())
+
+      return createPresignedPostDataPromise({
```

Comment:
hang on, we do want to specify the key here right (as uuid), instead of allowing the client to upload any file key